src/relations/postgresql_provider.py

```diff
@@ -73,6 +81,15 @@ def _sanitize_extra_roles(extra_roles: str | None) -> list[str]:
 
         return [role.lower() for role in extra_roles.split(",")]
 
+    def register_address_for_endpoint(self, _) -> None:
+        """Register the address for the endpoint.
+
+        If spaces are defined for endpoint, address are registered for the unit in peer relation.
+        """
+        address = str(self.model.get_binding(self.relation_name).network.bind_address)
+        self.charm.unit_peer_data[f"{self.relation_name}-address"] = address
```

Comment:
this will introduce a new field for all unit peer databag with current IP for this endpoint, e.g.:

```yaml
local-unit:
  in-scope: true
  data:
    database-address: 10.239.34.146 <--- this
    egress-subnets: 10.163.154.207/32
    exporter-started: "True"
    ingress-address: 10.163.154.207
    ...
```

The value is used to update endpoint for related apps, since patroni will not report the same addresses when using separated spaces for `peer` and `database`